src/main/java/io/aiven/kafka/connect/s3/config/AwsStsEndpointConfig.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.s3.config;
+
+import java.util.Objects;
+
+final class AwsStsEndpointConfig {
```

Comment:
Hm... I wonder: do we need a class to do a couple of null checks?

Reply:
I use it as a value object mainly: that is why I try to keep it immutable.  I like that all necessary fields that are used in same flow are grouped together.